devito/interfaces.py

```diff
@@ -1,9 +1,47 @@
 import numpy as np
 from sympy import IndexedBase
 from tools import aligned
+from hashlib import sha1
+from random import randint
+import os
 
 
 class DenseData(IndexedBase):
+
+    # hash function used for generating part of memmap file name
+    _hashing_function = sha1
+    # this is the directory where memmap files will be created
+    _disk_path = None
+    # holds the string name of all memmap file created
+    _memmap_file_list = []
+
+    ## functions for managing memmap files
+    # call this method to specify where you want memmap file to be created
+    @staticmethod
+    def set_disk_path(disk_path):
+        if not os.path.exists(disk_path):
+            print("the directory" + disk_path + " does not exit")
```

Comment:
No need to be so verbose about this. Creating path directories that don't exist is expected behaviour.

That being said, if we had a proper logging mechanism, this might be appropriate on a "debug" level, but otherwise it will become annoying to users quickly.